### PR TITLE
Add recipe for pocketsphinx-python

### PR DESCRIPTION
**Pull request by CurtLH**

### CODE REVIEW

File: recipes/pocketsphinx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{% set name = "pocketsphinx-python" %}
+{% set version = "0.1.3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/bambocher/{{ name }}/archive/v{{ version }}.tar.gz
```
Comment (mariusvniekerk) [inline recipes/pocketsphinx/meta.yaml]: could we get this from pypi instead?

Comment (CurtLH) [reply recipes/pocketsphinx/meta.yaml]: Sure.  When I first looked at [PyPI](https://pypi.org/project/pocketsphinx/#files), I only saw the `wheel` files, but upon closer inspection, I see the source tarball in there.

File: recipes/pocketsphinx/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/bambocher/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: d56244340161847cd978fdf9aba0d477bf5f99fcb3019f5b10dea921d078e23a 
+  - git_url: https://github.com/cmusphinx/pocketsphinx
+    git_tag: 1fdc9ccb637836d45d40956e745477a2bd3b470a
+    folder: deps/pocketsphinx
+  - git_url: https://github.com/cmusphinx/sphinxbase
```
Comment (scopatz) [inline recipes/pocketsphinx/meta.yaml]: Please do not use git for your source, see https://conda-forge.org/docs/meta.html#build-from-tarballs-not-repos

File: recipes/pocketsphinx/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    git_tag: 01392d23f8b45f4e0567c46057cf1149c9a124e7
```
Comment (scopatz) [inline recipes/pocketsphinx/meta.yaml]: Also, it doesn't seem like this has had a release....

File: recipes/pocketsphinx/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - swig
```
Comment (scopatz) [inline recipes/pocketsphinx/meta.yaml]: You might need to put `{{ compiler('c') }}` under host when using swig. Or maybe put swig under build

File: recipes/pocketsphinx/meta.yaml (hunk 3 above)
Comment (mariusvniekerk) [reply recipes/pocketsphinx/meta.yaml]: can still use a github archive

Comment (scopatz) [reply recipes/pocketsphinx/meta.yaml]: Yeah, definitely, just not a git url and tag

File: recipes/pocketsphinx/meta.yaml (hunk 2 above)
Comment (CurtLH) [reply recipes/pocketsphinx/meta.yaml]: How should I handle this situation if the upstream repo has not made a proper release?  I see their code on SurgeForce as well, but I'm not sure what version there matches to the commits of the packages under `deps`.

Comment (CurtLH) [reply recipes/pocketsphinx/meta.yaml]: I'll break this into separate recipes.   Thank you for your help!

Comment (scopatz) [reply recipes/pocketsphinx/meta.yaml]: No problem !

**Comment by CurtLH** [comment]
Thanks for taking a look @scopatz.  I accidentally triggered this branch instead of a different one.  I'm waiting on `pocketsphinx` to be merged first.

**Comment by scopatz** [comment]
Ahh gotcha :)

**Comment by CurtLH** [comment]
This build is failing because of the following reason:

```
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pip-3df3gcss-build/deps/pocketsphinx/model/en-us'
```

I believe this is a problem with [pocketsphinx](https://github.com/conda-forge/pocketsphinx-feedstock) not including the models.

**Comment by scopatz** [comment]
Ahh you may need to push up a new version of pocketsphinx then, or have a way of getting the models before building.  You may also consider having a separate models package.

File: recipes/pocketsphinx/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/bambocher/{{ name }}/archive/v{{ version }}.tar.gz
+    sha256: d56244340161847cd978fdf9aba0d477bf5f99fcb3019f5b10dea921d078e23a 
+  - url: https://sourceforge.net/projects/cmusphinx/files/pocketsphinx/5prealpha/pocketsphinx-5prealpha.tar.gz
+    sha256: ef5bb5547e2712bdf571f256490ef42a47962033892efd9d7df8eed7fe573ed9
+    folder: deps
```
Comment (sodre) [inline recipes/pocketsphinx/meta.yaml]: try `deps/pocketsphinx`

File: recipes/pocketsphinx/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  - url: https://sourceforge.net/projects/cmusphinx/files/sphinxbase/5prealpha/sphinxbase-5prealpha.tar.gz
+    sha256: f72bdb59e50b558bed47cc2105777200d2b246a0f328e913de16a9b22f9a246f
+    folder: deps
```
Comment (sodre) [inline recipes/pocketsphinx/meta.yaml]: try `deps/sphinxbase`

**Comment by CurtLH** [comment]
According to the [pocketsphinx-python GitHub README](Python interface to CMU SphinxBase and PocketSphinx libraries created with SWIG. Pocketsphinx packages include python support, however, it is based on Automake and not well supported on Windows.)...

> Python interface to CMU SphinxBase and PocketSphinx libraries created with SWIG. Pocketsphinx packages include python support, however, it is based on Automake and not well supported on Windows.

For this reason, I've decided to skip the Windows build for now.

**Comment by CurtLH** [comment]
This one is ready for review @conda-forge/staged-recipes

File: recipes/pocketsphinx/yum_requirements.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+pulseaudio-libs-devel
```
Comment (scopatz) [inline recipes/pocketsphinx/yum_requirements.txt]: It would be nice to have this as an actual conda-forge package.

**Comment by scopatz** [comment]
Thanks @CurtLH!

File: recipes/pocketsphinx/yum_requirements.txt (hunk 7 above)
Comment (CurtLH) [reply recipes/pocketsphinx/yum_requirements.txt]: I asked about one [Gitter](https://gitter.im/conda-forge/conda-forge.github.io?at=5b637fd5945df30dc1462060), and it seemed like the response was that `pulseaudio` would be a pain to package.  I'm sure it's possible, but I'm also sure that it's over my head! 🙂

Comment (scopatz) [reply recipes/pocketsphinx/yum_requirements.txt]: totally fair!

**Pull request merged**